functorch/_src/vmap.py

```diff
@@ -352,18 +352,139 @@ def vmap(
         vmap does not provide general autobatching or handle variable-length
         sequences out of the box.
     """
-    if randomness not in ['error', 'different', 'same']:
-        raise RuntimeError(f"Only allowed values for randomness are 'error', 'different', or 'same'. Got {randomness}")
+    _check_randomness_arg(randomness)
 
     @functools.wraps(func)
     def wrapped(*args, **kwargs):
         _check_out_dims_is_int_or_int_pytree(out_dims, func)
         batch_size, flat_in_dims, flat_args, args_spec = _process_batched_inputs(in_dims, args, func)
-        vmap_level = _vmap_increment_nesting(batch_size, randomness)
-        try:
-            batched_inputs = _create_batched_inputs(flat_in_dims, flat_args, vmap_level, args_spec)
-            batched_outputs = func(*batched_inputs, **kwargs)
-            return _unwrap_batched(batched_outputs, out_dims, vmap_level, batch_size, func)
-        finally:
-            _vmap_decrement_nesting()
+        return _flat_vmap(
+            func, batch_size, flat_in_dims, flat_args, args_spec, out_dims, randomness, **kwargs
+        )
+
     return wrapped
+
+
+def chunk_vmap(
+        func: Callable,
+        in_dims: in_dims_t = 0,
+        out_dims: out_dims_t = 0,
+        randomness: str = 'error',
+        chunks=2) -> Callable:
+    """
+    chunk_vmap is the vectorizing map using chunks of input data. Splitting data into chunks
+    can help to prevent out-of-memory issues. For more details about vectorizing map, see :func:`vmap`.
```

Comment:
nit: (I can also write this later if you don't have bandwidth). It might be easier to explain `chunk_vmap` as a mix between vmap (which vectorizes everything) and map (which executes things sequentially). `chunk_vmap` vectorize things `#chunks` at a time.